### ISSUE TITLE
Files are compiled with the wrong plugin

### ISSUE THREAD

**Issue by fubuloubu** [ApeWorX/ape#1031]
### Environment information

```
Cache restored from key: 3.8-pip-
Run pip install eth-ape
Collecting eth-ape
  Using cached eth_ape-0.4.5-py3-none-any.whl (180 kB)
```

```sh
$ ape --version
# ...copy and paste result of above command here...

ape-vyper 0.4.1
ape-solidity 0.4.1

$ ape plugins list
# ...copy and paste result of above command here...
```

* Contents of your `ape-config.yaml` (NOTE: do not post anything private like RPC urls or secrets!):

```sh
$ cat ape-config.yaml
# ...copy and paste result of above command here...
```

### What went wrong?

If a project has multiple languages, current logic is trying to compile contracts with the wrong compiler plugin
https://github.com/yearn/veYFI/runs/8177605680

### How can it be fixed?

Some sort of regression with latest compiler manager orchestration logic in `ProjectAPI`

**Comment by unparalleled-js** [comment]
I thought we fixed this in 0.4.5. I noticed the  build output says it was using a cached 0.4.5. Hmmmmm......

When digging into this problem, I noticed a separate issue regarding compiling yearn vaults and fixed it here: https://github.com/ApeWorX/ape-solidity/pull/76

**Comment by fubuloubu** [comment]
@unparalleled-js is this still an issue?

**Comment by unparalleled-js** [comment]
> @unparalleled-js is this still an issue?

The issue was fixed in 0.4.5
I installed 0.4.5 and all the same plugin versions from the action output and things worked.
I believe there is a GH caching issue of some sort.
